File: lib/features/notices/data/models/notice.dart
class Notice {
  final String id;
  final String title;
  final String category; // Academic, Events, Placement, Administrative
  final String date;
  final String snippet;
  final String description;
  final String imageUrl;

  const Notice({
    required this.id,
    required this.title,
    required this.category,
    required this.date,
    required this.snippet,
    required this.description,
    required this.imageUrl,
  });

  Notice copyWith({
    String? id,
    String? title,
    String? category,
    String? date,
    String? snippet,
    String? description,
    String? imageUrl,
  }) {
    return Notice(
      id: id ?? this.id,
      title: title ?? this.title,
      category: category ?? this.category,
      date: date ?? this.date,
      snippet: snippet ?? this.snippet,
      description: description ?? this.description,
      imageUrl: imageUrl ?? this.imageUrl,
    );
  }
}

File: backend/src/models/Notice.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Notice = sequelize.define('Notice', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  snippet: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  category: {
    type: DataTypes.ENUM('Academic', 'Events', 'Placement', 'Administrative'),
    allowNull: false,
    defaultValue: 'Academic',
  },
  imageUrl: {
    type: DataTypes.STRING,
    allowNull: true,
  },
});

export default Notice;
